x-pack/plugins/security_solution/public/common/components/hover_actions/use_hover_action_items.tsx

```diff
@@ -105,6 +108,40 @@ export const useHoverActionItems = ({
     [uiSettings, timelineId, activeFilterManager, filterManagerBackup]
   );
 
+  /*
+   *   Add to Timeline button, adds data to dataprovider but does not persists the Timeline
+   *   to the server because of following reasons.
+   *
+   *   1. Add to Timeline button performs actions in `timelines` plugin
+   *   2. `timelines` plugin does not have information on how to create/update the timelines in the server
+   *       as it is owned by Security Solution
+   * */
+  const OnAddToTimeline = useCallback(() => {
+    if (!dataProvider || isEmpty(dataProvider)) return;
+    if (dataProvider instanceof Array) {
+      dataProvider.forEach((provider) => {
```

Comment:
instead of a loop here for each data provider, can we change the payload to accept either a single provider or an array of them?

Reply:
@kqualters-elastic @michaelolo24 , I pushed the changes, but I would like to ask if there are any scenarios when there are multiple `dataProviders` added at once because I was not able find them.

Reply:
If you add any of the `source.ip` or `destination.ip` fields for instance